scipy/optimize/constraints.py

```diff
@@ -0,0 +1,617 @@
+from __future__ import division, print_function, absolute_import
+import numpy as np
+import scipy.sparse as spc
+
+
+__all__ = ['CanonicalConstraint',
+           'NonlinearConstraint',
+           'LinearConstraint',
+           'BoxConstraint',
+           'check_sparsity',
```

Comment:
Ok, I change it to 

      from ._constraints import (NonlinearConstraint, LinearConstraint, BoxConstraint)